### PR TITLE
Updating bubs-static to use Eleventy

### PR DESCRIPTION
**Pull request by chrisherold**
Use eleventy static site generator with bubs-style gulp workflow and sass

### CODE REVIEW

File: src/site/_includes/layouts/base.njk [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+      <header>
+        <ul class="nav">
+          <li>
+            <a href="/">home<small>where the heart is</small></a>
+          </li>
+          <li>
+            <a href="/about">about<small>all about this</small></a>
+          </li>
+        </ul>
+        <h1>{{ title }}</h1>
```
Comment (imjared) [inline src/site/_includes/layouts/base.njk]: can remove this or the title in `post.njk`

File: src/site/_includes/layouts/post.md [hunk 2]
```diff
@@ -0,0 +1,18 @@
+---
+layout: layouts/base.njk
+pageClass: posts
+templateEngineOverride: njk, md
+---
+
+<h1>{{ title }}</h1>
```
Comment (imjared) [inline src/site/_includes/layouts/post.md]: can remove either this or the title in `layouts/base.njk`

**Comment by imjared** [comment]
1. Can we provide an example of extending the default layout with simple HTML rather than `.md`? eg. `foo.njk`:
    ```
    {% extends "layouts/base.njk" %}

    {% block content %}
    <h1>here is a test</h1>
    {% endblock %}
    ```

1. Can we include a note in `readme.md` about having gulp [installed as a global dep](https://github.com/gulpjs/gulp-cli/issues/84#issuecomment-272958709)?

**Comment by imjared** [comment]
upon further inspection/usage:
- would be nice to clear out the old stuff. no real need for the old gulpfile
- [gulp-notify](https://www.npmjs.com/package/gulp-notify) or [node-notifier](https://www.npmjs.com/package/node-notifier) would be really helpful to provide alerts for CSS/JS compilation

File: gulpfile.old [hunk 3]
```diff
@@ -0,0 +1,163 @@
+'use strict';
```
Comment (imjared) [inline gulpfile.old]: can delete this whole file

File: package.json [hunk 4]
```diff
@@ -1,52 +1,50 @@
...
+  "license": "MIT",
   "scripts": {
-    "start": "node index.js"
   },
-  "dependencies": {},
+  "repository": {
+    "type": "git",
+    "url": "https://github.com/patronage/bubs-static"
+  },
+  "dependencies": {
```
Comment (imjared) [inline package.json]: all the gulp pieces are dev deps

File: src/scss/_legacy.scss [hunk 5]
```diff
@@ -0,0 +1,183 @@
+* {
```
Comment (imjared) [inline src/scss/_legacy.scss]: can remove this file

File: gulpfile.old [hunk 6]
```diff
@@ -0,0 +1,163 @@
...
+      .pipe($.autoprefixer('last 2 versions'))
+      .pipe($.sourcemaps.write(sourcemapsOptions))
+      .pipe($.if(isProduction, $.csso()))
+      .pipe(gulp.dest( config.output + '/css' ))
+      .pipe(browserSync.stream());
+}
+
+function scripts() {
+  return gulp.src( config.assets + "/js/**/*.js")
+      .pipe($.concat('main.js'))
```
Comment (imjared) [inline gulpfile.old]: this needs to be fixed from the previous version. it makes it impossible for scripts to rely on each other. i'm using `fullpage.js` and it's [very specific about the order that scripts are called](https://github.com/alvarotrigo/fullPage.js#use-extensions).

for the time being, i can alphabetize my scripts but we should use something like [gulp-html-replace](https://www.npmjs.com/package/gulp-html-replace) to ensure order

**Comment by chrisherold** [comment]
@imjared on the gulp error `Failed to load external module @babel/register`

`Not an issue, as it states on the next line down, it required babel-core/register. Babel now suggests you use a module called babel-register so we try that first with the fallback.` https://github.com/gulpjs/gulp/issues/1631#issuecomment-216332285

File: gulpfile.old (hunk 6 above)
Comment (imjared) [reply gulpfile.old]: it's also possible i'm not using this correctly. if you have another rec for including plugins, i'm all ears

Comment (chrisherold) [reply gulpfile.old]: i don't like how this works either, it's just trying to shove everything into one js file and doesn't really have good support for plugins. i'm working on a solution for this now.

**Comment by chrisherold** [comment]
think i got everything above in this latest commit

**Comment by imjared** [comment]
would be really nice if we could get some es6 action in here for scripts: https://www.npmjs.com/package/gulp-babel

**Pull request merged**